packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx

```diff
@@ -0,0 +1,173 @@
+  const { getColumnDisplayName } = useQueryVisualizationContext()
+
+  const currentSelectedFacet: FacetColumnValuesRequest | undefined =
+    useMemo(() => {
+      const facetColumnRequest = getCorrespondingSelectedFacet(
+        facet,
+        cloneDeep(nextQueryRequest.query.selectedFacets) as
+          | FacetColumnRequest[]
+          | undefined,
+      )
+
+      if (
+        facetColumnRequest &&
+        !isFacetColumnValuesRequest(facetColumnRequest)
+      ) {
+        console.error(
+          `The facet rendered in EnumFacetFilter is not a FacetColumnValuesRequest`,
+          facet,
+        )
+        return undefined
+      }
+      return facetColumnRequest
+    }, [facet, nextQueryRequest.query.selectedFacets])
+
+  // Must compare the known facet values to the "uncommitted" current query
+  const allIsSelected = getAllIsSelected(
+    cloneDeep(nextQueryRequest) as QueryBundleRequest,
+    facet,
+  )
+
+  const columnModel = data?.columnModels
+    ? getCorrespondingColumnForFacet(facet, data.columnModels)
+    : undefined
+
+  const userIds =
+    columnModel?.columnType === ColumnTypeEnum.USERID ||
+    columnModel?.columnType === ColumnTypeEnum.USERID_LIST
+      ? facet.facetValues.map(facet => facet.value)
+      : []
+  const userGroupHeaders = useGetInfoFromIds<UserGroupHeader>({
+    ids: userIds,
+    type: 'USER_PROFILE',
+  })
+
+  const entityIds =
+    columnModel?.columnType === ColumnTypeEnum.ENTITYID ||
+    columnModel?.columnType === ColumnTypeEnum.ENTITYID_LIST
+      ? facet.facetValues.map(facet => facet.value)
+      : []
+  const entityHeaders = useGetInfoFromIds<EntityHeader>({
+    ids: entityIds,
+    type: 'ENTITY_HEADER',
+  })
+
+  const evaluationIds =
+    columnModel?.columnType === ColumnTypeEnum.EVALUATIONID
+      ? facet.facetValues.map(facet => facet.value)
+      : []
+  const evaluations = useGetInfoFromIds<Evaluation>({
+    ids: evaluationIds,
+    type: 'EVALUATION_QUEUE',
+  })
+
+  const displayedFacetValues: RenderedFacetValue[] = useMemo(() => {
+    return sortBy(
+      facet.facetValues.map(
+        (facetValue: FacetColumnResultValueCount): RenderedFacetValue => {
+          return {
+            ...facetValue,
+            // Selected status should be based on the 'nextQuery', not the result data
+            // This ensures the checkboxes respond instantly to user interaction, like while waiting for multiple changes to debounce
+            isSelected:
+              currentSelectedFacet?.facetValues.includes(facetValue.value) ??
+              false,
+            displayText: valueToLabel(
+              facetValue,
+              userGroupHeaders,
+              entityHeaders,
+              evaluations,
+            ),
+          }
+        },
+      ),
+      fv => -1 * fv.count,
+    )
+  }, [
+    currentSelectedFacet?.facetValues,
+    entityHeaders,
+    evaluations,
+    facet.facetValues,
+    userGroupHeaders,
+  ])
```

Comment:
Transform all of our facet data into a list of a custom type `RenderedFacetValue`, which is what EnumFacetFilterUI uses for display